examples/rn-todo-list/frontend/app/AppSync.tsx

```diff
@@ -0,0 +1,89 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import React from 'react';
+import {SafeAreaView, StyleSheet} from 'react-native';
+import {OpenRealmBehaviorType} from 'realm';
+import {AppProvider, RealmProvider, UserProvider} from '@realm/react';
+
+import {LoginScreen} from './screens/LoginScreen';
+import {Task} from './models/Task';
+import {TaskScreenSync} from './screens/TaskScreenSync';
+import {colors} from './styles/colors';
+import {schemas} from './models';
+
+type AppSyncProps = {
+  appId: string;
+};
+
+/**
+ * The root React component for the Device Sync enabled app which renders
+ * `@realm/react`'s `AppProvider` for instantiation an Atlas App Services
+ * App, `UserProvider` for providing the App User once authenticated, and
+ * `RealmProvider` for opening a Realm.
+ */
+export function AppSync({appId}: AppSyncProps) {
+  return (
+    <SafeAreaView style={styles.screen}>
+      <AppProvider id={appId}>
+        {/* The component set as the `fallback` prop will be rendered if a user has
+        not been authenticated. In this case, we will show the login screen. */}
+        <UserProvider fallback={<LoginScreen />}>
+          {/* Define the Realm configuration as props passed to `RealmProvider`.
+          Note that `user` does not need to be defined in the `sync` config
+          since the `RealmProvider` will set it for you once authenticated. */}
+          <RealmProvider
+            schema={schemas}
+            sync={{
+              flexible: true,
+              // To sync data to the device, we need to subscribe to our tasks.
+              initialSubscriptions: {
+                update: (mutableSubs, realm) =>
+                  mutableSubs.add(realm.objects(Task), {name: 'myTasks'}),
+              },
+              // We can specify the behavior when opening a Realm for the first time
+              // (`newRealmFileBehavior`) and for subsequent ones (`existingRealmFileBehavior`).
+              // If the user has logged in at least 1 time before, the Realm and its data will
+              // exist on disk and can be opened even when offline. We can either (a) open the
+              // Realm immediately (or first create a new empty Realm file if it does not
+              // exist before opening it) and sync the data to the device in the background
+              // (`OpenRealmBehaviorType.OpenImmediately`), or (b) wait for any non-synced
+              // data to be fully downloaded (`OpenRealmBehaviorType.DownloadBeforeOpen`).
+              // For more possible configurations of new and existing Realm file behaviors, see:
+              // https://www.mongodb.com/docs/realm-sdks/js/latest/types/OpenRealmBehaviorConfiguration.html
+              newRealmFileBehavior: {
+                type: OpenRealmBehaviorType.OpenImmediately,
```

Comment:
Ah, forgot to look over these settings.  I think I'd prefer `DownloadBeforeOpen` for the `newRealmFileBehavior`.  Otherwise it will start with an empty list and suddenly a populated list will appear (if there was something pre-existing).

Reply:
Yeah, since the first open would also have required the user to be online from logging in, that may be better 👍 Will test how that looks. Not easy at this point though to show some decent loading until we have the progress notifications project done right?

Reply:
Just the a simple spinner at this point

Reply:
`RealmProvider.fallback` to the rescue 🙏